src/pages/MyContributions/index.tsx

```diff
@@ -0,0 +1,69 @@
+import { gql } from "@apollo/client";
+import { useMemo } from "react";
+import PaymentTable from "src/components/Payments";
+import QueryWrapper from "src/components/QueryWrapper";
+import { useAuth } from "src/hooks/useAuth";
+import { useHasuraQuery } from "src/hooks/useHasuraQuery";
+import { Currency, HasuraUserRole, Payment, PaymentStatus } from "src/types";
+
+const MyContributions = () => {
+  const { user } = useAuth();
+  const query = useHasuraQuery(GET_MY_CONTRIBUTIONS_QUERY, HasuraUserRole.User, {
+    variables: { userId: user?.id },
+  });
+  const { data } = query;
+  const payments = useMemo(() => data?.payment_requests?.map(mapApiPaymentsToProps) ?? null, [data]);
```

Comment:
Not sure useMemo is useful here, as the data is the only state variable that is going to change often in this component. If user / query changes we probably want to rerender the component and the payments.
Also, I think useMemo only does a shallow comparison for the data in its second parameter, so the component wouldn't get rerendered if some properties from the query result change.